addons/membership/models/account_invoice.py

```diff
@@ -60,9 +60,9 @@ def create(self, vals):
                 invoice_line.product_id.membership and \
                 not MemberLine.search([('account_invoice_line', '=', invoice_line.id)]):
             # Product line is a membership product
-            date_from = invoice_line.product_id.membership_date_from or '0000-00-00'
-            date_to = invoice_line.product_id.membership_date_to or '0000-00-00'
-            if date_from < (invoice_line.invoice_id.date_invoice or '0000-00-00') < date_to:
+            date_from = invoice_line.product_id.membership_date_from
+            date_to = invoice_line.product_id.membership_date_to
+            if date_from and date_from < (invoice_line.invoice_id.date_invoice or '0000-00-00') < (date_to or '0000-00-00'):
```

Comment:
What is the point of this change?

Reply:
Otherwise if `date_from` or `date_false` are `False` this comparison is going to fail in python3.

With this assignment it's going to fail also when it tries to write a '0000-00-00' date in the db:
```python
date_from = invoice_line.product_id.membership_date_from or '0000-00-00'
date_to = invoice_line.product_id.membership_date_to or '0000-00-00'
```

Reply:
> With this assignment it's going to fail also when it tries to write a '0000-00-00' date in the db

Right, that's the part I was wondering about, makes sense.